libwaysip/src/lib.rs

```diff
@@ -35,12 +35,35 @@ fn get_cursor_buffer(connection: &Connection, shm: &WlShm) -> Option<CursorImage
     Some(cursor?[0].clone())
 }
 
+pub struct WaysipConnection<'a> {
+    pub connection: &'a Connection,
+    pub globals: GlobalList,
+}
+
 /// get the selected area
 pub fn get_area(
-    connection: &Connection,
-    globals: GlobalList,
+    waysip_connection: Option<WaysipConnection>,
     selection_type: SelectionType,
 ) -> Result<Option<state::AreaInfo>, WaySipError> {
+    let (connection, globals) = match waysip_connection {
+        Some(WaysipConnection {
+            connection,
+            globals,
+        }) => (connection.clone(), globals),
+        None => {
+            let connection =
+                Connection::connect_to_env().map_err(|e| WaySipError::InitFailed(e.to_string()))?;
+            let (globals, _) = registry_queue_init::<state::WaysipState>(&connection)
+                .map_err(|e| WaySipError::InitFailed(e.to_string()))?; // We just need the
```

Comment:
Comment is outdated. BaseState no longer exists. We can just delete this comment.

Reply:
done